pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go

```diff
@@ -230,23 +241,49 @@ func TestPipelineRunGetServiceAccountName(t *testing.T) {
 	}{
 		{
 			name: "default SA",
-			pr: tb.PipelineRun("pr",
-				tb.PipelineRunSpec("prs",
-					tb.PipelineRunServiceAccountName("defaultSA"),
-					tb.PipelineRunServiceAccountNameTask("taskName", "taskSA"))),
+			pr: &v1alpha1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pr",
+				},
+				Spec: v1alpha1.PipelineRunSpec{
+					PipelineRef: &v1alpha1.PipelineRef{
+						Name: "prs",
+					},
+					ServiceAccountName: "defaultSA",
+					ServiceAccountNames: []v1alpha1.PipelineRunSpecServiceAccountName{{
+						TaskName:           "taskName",
+						ServiceAccountName: "taskSA",
+					}},
+				},
+			},
 			saNames: map[string]string{
 				"unknown":  "defaultSA",
 				"taskName": "taskSA",
 			},
 		},
 		{
 			name: "mixed default SA",
-			pr: tb.PipelineRun("defaultSA",
-				tb.PipelineRunSpec("defaultSA",
-					tb.PipelineRunServiceAccountName("defaultSA"),
-					tb.PipelineRunServiceAccountNameTask("task1", "task1SA"),
-					tb.PipelineRunServiceAccountNameTask("task2", "task2SA"),
-				)),
+			pr: &v1alpha1.PipelineRun{
```

Comment:
Hey @abayer I appreciate all the cleanup here 🙏  Can I also request to utilize the `YAML parser` helpers defined [here](https://github.com/tektoncd/pipeline/blob/00b55df87e4815ac0647e53fb11f242e28ff2f45/test/yaml.go#L51 ) if its possible instead of initiating `structs`?  What do other folks think?

Reply:
I'm personally not a big fan of embedding YAML in the tests like that, but I'm not _opposed_ to it per se. Lemme finish up removing `internal/builder` completely and then let's have a follow-up issue for the YAML approach?

Reply:
sure sounds good, thanks @abayer 👍

Reply:
(spoiler: I'll probably end up doing the move to YAML approach if there's a consensus on that change, since I've got a good chunk of free time, but I'm so dang close to finishing removing `internal/builder` that I want to put that to bed first. =) )